suite-native/device/src/hooks/useDeviceConnect.tsx

```diff
@@ -0,0 +1,93 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+
+import {
+    ConnectDeviceStackParamList,
+    ConnectDeviceStackRoutes,
+    RootStackParamList,
+    RootStackRoutes,
+    StackToStackCompositeNavigationProps,
+} from '@suite-native/navigation';
+import {
+    acquireDevice,
+    selectDeviceRequestedPin,
+    selectIsDeviceConnectedAndAuthorized,
+    selectIsDeviceConnectedButNotInitialized,
+    selectIsUnacquiredDevice,
+} from '@suite-common/wallet-core';
+import { useAlert } from '@suite-native/alerts';
+import { useTranslate } from '@suite-native/intl';
+
+type NavigationProps = StackToStackCompositeNavigationProps<
+    ConnectDeviceStackParamList,
+    ConnectDeviceStackRoutes.ConnectDeviceCrossroads,
+    RootStackParamList
+>;
+
+export const useDeviceConnect = () => {
+    const hasDeviceRequestedPin = useSelector(selectDeviceRequestedPin);
+    const isDeviceConnectedAndAuthorized = useSelector(selectIsDeviceConnectedAndAuthorized);
+    const isUnacquiredDevice = useSelector(selectIsUnacquiredDevice);
+    const isDeviceConnectedButNotInitialized = useSelector(
+        selectIsDeviceConnectedButNotInitialized,
+    );
+
+    const dispatch = useDispatch();
+
+    const navigation = useNavigation<NavigationProps>();
+
+    const { hideAlert, showAlert } = useAlert();
+
+    const { translate } = useTranslate();
+
+    useEffect(() => {
+        if (hasDeviceRequestedPin) {
+            navigation.navigate(RootStackRoutes.ConnectDevice, {
+                screen: ConnectDeviceStackRoutes.PinMatrix,
+            });
+        }
+    }, [hasDeviceRequestedPin, navigation]);
+
+    // If device is unacquired (restarted app, another app fetched device session, ...),
+    // we cannot work with device anymore. Shouldn't happen on mobile app but just in case.
+    useEffect(() => {
+        if (isUnacquiredDevice) {
+            showAlert({
+                title: translate('moduleConnectDevice.unacquiredDeviceModal.title'),
+                description: translate('moduleConnectDevice.unacquiredDeviceModal.description'),
+                icon: 'warningCircle',
+                pictogramVariant: 'red',
+                primaryButtonTitle: translate('moduleConnectDevice.unacquiredDeviceModal.button'),
+                onPressPrimaryButton: () => dispatch(acquireDevice()),
+            });
+        } else {
+            hideAlert();
```

Comment:
the `hideAlert` will hide any alert that is displayed at the moment. Could not happen that this `hideAlert()` call will hide the modal triggered from other useEffect working with `isDeviceConnectedButNotInitialized` when the timing is not ideal?

Not sure if that could happen just thinking out loud.

Reply:
Yes, but since this modal takes precedence in terms of we need to have running session to communicate with the device, then I want the other modal to be gone for this one to appear. Then if we get the session, then we can read from device again to display info about device not being initialized.

Don't worry about this too much as this will only happen in development with `trezor-user-env` but on mobile is nearly impossible to run into this scenario.